### PR TITLE
Add RDF analysis

### PR DESCRIPTION
**Pull request by jennyfothergill**
begins to address #4 

(not sure why, but perhaps the shift from public to private borked my previous PR #23) 

- [x] add units to plot and docstring
- [x] add unit test

### CODE REVIEW

File: project/src/analysis/rdf.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import gsd.hoomd
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def gsd_rdf(job):
+    """Compute the RDF given a GSD file."""
+    gsdfile = job.fn("trajectory.gsd")
+
+    # TODO check that these are reasonable values
```
Comment (justinGilmer) [inline project/src/analysis/rdf.py:14]: Might want to make these function arguments with these as the default values?

**Review by justinGilmer — APPROVED** [review]
Small comment, but looks good overall!

File: project/src/analysis/rdf.py (hunk 1 above)
Comment (jennyfothergill) [reply project/src/analysis/rdf.py:14]: just to make sure I understand--do you mean something like
```python
def gsd_rdf(job, fname="trajectory.gsd"):
    """Compute the RDF given a GSD file."""
    gsdfile = job.fn(fname)
```
?

File: project/src/analysis/rdf.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    frames = 10
+    bins = 50
+    r_min = 0.5
+    r_max = 2.5
```
Comment (ramanishsingh) [inline project/src/analysis/rdf.py:18]: Shall we change r_max to the (minimum (list of side length in each prod traj frame))/2-0.01 ?

File: project/src/analysis/rdf.py (hunk 1 above)
Comment (jennyfothergill) [reply project/src/analysis/rdf.py:14]: On further thought, I think you mean
```
def gsd_rdf(job, frames = 10, bins = 50, r_min = 0.5, r_max = 2.5):
```
and I agree!

Comment (justinGilmer) [reply project/src/analysis/rdf.py:14]: Yeah this is what I meant!

File: project/src/analysis/rdf.py (hunk 2 above)
Comment (jennyfothergill) [reply project/src/analysis/rdf.py:18]: As the longest possible r_max, that is a good place to start! Depending on how big our systems get, we may want to go smaller, but I agree that we should have this as the default.

Comment (ramanishsingh) [reply project/src/analysis/rdf.py:18]: I think freud complains when r_max > L/2. But we can confirm that once we start the analysis.

Comment (jennyfothergill) [reply project/src/analysis/rdf.py]: Yes, you are correct. I have updated this code to use a default r_max based on box lengths
(minimum box length * 0.45)

**Review by ramanishsingh — APPROVED** [review]

**Comment by chrisjonesBSU** [comment]
> LGTM, but one thought. Should the `run_analysis` operation have some requirement that the simulation is finished rather than just checking that there is a `trajectory.gsd` file?

I guess I'm not considering the step that sanitizes the outputs (#25 #28), which would occur after a simulation finishes and before `run_analysis`.  So, maybe checking that a simulation is finished isn't required here.

**Comment by jennyfothergill** [comment]
> > LGTM, but one thought. Should the `run_analysis` operation have some requirement that the simulation is finished rather than just checking that there is a `trajectory.gsd` file?
> 
> I guess I'm not considering the step that sanitizes the outputs (#25 #28), which would occur after a simulation finishes and before `run_analysis`. So, maybe checking that a simulation is finished isn't required here.

yeah, exactly. Because each output should be sanitized before analysis, the trajectory file existing is a proxy for the simulation being finished. 
Once we add equilibration/decorrelation to our workflows that may be added as a precondition.

**Comment by jennyfothergill** [comment]
This should be ready for review. I'd like feedback on how to improve the unit test.

**Comment by justinGilmer** [comment]
Overall this looks good, but maybe we can add in a RDF of a system we can verify through another calculation? Like make a similar file output from a xtal system and a liq system? We can change the methods up a bit like this to make it easier

```python3
def gsd_rdf(job, frames=10, stride=1, bins=50, r_min=0.5, r_max=None):
    return _gsd_rdf_(job.trj, ...)

def _gsd_rdf_(trj, frames, stride, bins, r_min, r_max):
    # calculations here
```

**Comment by justinGilmer** [comment]
Then we can test the output of the method without relying on a job at first.

**Review by justinGilmer — CHANGES_REQUESTED** [review]
Overall LGTM, i added one small suggestion to maybe add a known rdf and gsd file to then compare the output. Sort of hacky, but i cant really think of a way to test the rdf more.

File: project/tests/base_test.py [hunk 3]
```diff
@@ -13,7 +17,50 @@ def spceff(self, name="spce.xml"):
...
+    def tmp_project(self):
+        with signac.TemporaryProject(name="test") as p:
+            return p
+
+    @pytest.fixture
+    def tmp_job(self, tmp_project, statepoint={"a": 0}):
+        tmp_project.open_job(statepoint).init()
+        for job in tmp_project:
+            return job
+
```
Comment (justinGilmer) [inline project/tests/base_test.py:35]: This needs the `@pytest.fixture` decorator

**Review by justinGilmer — CHANGES_REQUESTED** [review]

File: project/tests/base_test.py (hunk 3 above)
Comment (justinGilmer) [reply project/tests/base_test.py:35]: This is for the `trappe_ua` fixture. It did not select multiple lines. Whoops.

Comment (jennyfothergill) [reply project/tests/base_test.py:34]: yeah, that's my bad! I've fixed it in a later commit.

**Comment by jennyfothergill** [comment]
@justinGilmer thank you for your suggestions! I've taken a shot at implementing them.

**Review by justinGilmer — APPROVED** [review]
This is excellent @jennyfothergill ! (I was unaware of the `TemporaryProject` ability too! 

This LGTM!

**Pull request merged**